www/include/options/accessLists/resourcesACL/DB-Func.php

```diff
@@ -622,3 +629,68 @@ function updateMetaServices($acl_id = null)
         }
     }
 }
+
+/**
+ * sanitize resources parameter for Create / Update a Resource ACL
+ *
+ * @param array<string, mixed> $resources
+ * @return array<string, mixed>
+ */
+function sanitizeResourceParameters(array $resources): array
+{
+    $sanitizedParameters = [];
+
+    $sanitizedParameters['acl_res_name'] = filter_var($resources['acl_res_name'], FILTER_SANITIZE_STRING);
+    $sanitizedParameters['acl_res_alias'] = filter_var($resources['acl_res_alias'], FILTER_SANITIZE_STRING);
+    $sanitizedParameters['acl_res_comment'] = filter_var($resources['acl_res_comment'], FILTER_SANITIZE_STRING);
+
+    if (
+        isset($resources['all_hosts']['all_hosts'])
+        && filter_var($resources['all_hosts']['all_hosts'], FILTER_VALIDATE_INT) !== false
+    ) {
+        /**
+         * Cast to string as it will be inserted as an enum '0','1'
+         */
+        $sanitizedParameters['all_hosts'] = (string) $resources['all_hosts']['all_hosts'];
+    } else {
+        $sanitizedParameters['all_hosts'] = "0";
+    }
```

Comment:
```suggestion
// Cast to string as it will be inserted as an enum '0','1'
$sanitizedParameters['all_hosts'] = (string) filter_var(
    $resources['all_hosts']['all_hosts'] ?? null,
    FILTER_VALIDATE_INT,
    [
        "options" => [
            "default" => 0
        ]
    ]
);

```